### PR TITLE
dashboard: use context api

### PR DESCRIPTION
**Pull request by AdirAmsalem**
Remove `@wix/dashboard-react` and `@wix/sdk-react` in favor of the context api.

### CODE REVIEW

File: custom-products-catalog/template/src/dashboard/withProviders.tsx [hunk 1]
```diff
@@ -6,9 +6,9 @@ import { useEnvironment } from '@wix/sdk-react';
 
 export function withProviders<P extends {} = {}>(Component: React.FC<P>) {
   return withDashboard(function DashboardProviders(props: P) {
-    const { locale } = useEnvironment<EnvironmentState>();
+    const { language } = useEnvironment<EnvironmentState>();
     return (
-      <WixDesignSystemProvider locale={locale} features={{ newColorsBranding: true }}>
+      <WixDesignSystemProvider locale={language} features={{ newColorsBranding: true }}>
         <WixPatternsProvider>
```
Comment (AdirAmsalem) [inline custom-products-catalog/template/src/dashboard/withProviders.tsx:12]: `@wix/patterns` are still relying on `@wix/dashboard-react` & `@wix/sdk-react` so we can't remove `withDashboard` just yet.

File: site-popup/template/src/dashboard/pages/page.tsx [hunk 2]
```diff
@@ -22,11 +21,14 @@ function SitePopup() {
...
     useEmbeds<Partial<SitePopupOptions>>();
 
+  const [ language, setLanguage ] = useState('');
   const [sitePopupOptions, setSitePopupOptions] = useState<SitePopupOptions>(
     sitePopupDefaultOptions
   );
 
-  const { locale } = useEnvironment<EnvironmentState>();
+  useEffect(() => {
+    dashboard.getLanguage().then((lang) => setLanguage(lang));
```
Comment (fantua) [inline site-popup/template/src/dashboard/pages/page.tsx:30]: This is not documented API. Not sure we can use it cc @dor-itzhaki @netanelgilad

File: custom-products-catalog/template/src/dashboard/withProviders.tsx (hunk 1 above)
Comment (netanelgilad) [reply custom-products-catalog/template/src/dashboard/withProviders.tsx:10]: btw, do we know if wix design system actually needs language or locale? see https://docs.google.com/document/d/129CGwbxpLFMOvvvLww9CXn0v3nh1EcJk3lm6K42GpO0/edit?usp=sharing

File: site-popup/template/src/dashboard/pages/page.tsx (hunk 2 above)
Comment (netanelgilad) [reply site-popup/template/src/dashboard/pages/page.tsx:30]: This is moving to `i18n` in `@wix/essentials`. Soon to be released and documented (already talked with the TW and it's going into their upcoming sprint).

**Review by fantua — APPROVED** [review]

**Pull request merged**